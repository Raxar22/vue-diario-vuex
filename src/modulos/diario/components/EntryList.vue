<template>
    <div class="entry-list-container">
        <div class="px-2 pt-2">
            <input  type="text"
                    class="form-control"
                    placeholder="Buscar Entrada"
                    v-model="term"/>
        </div>

        <div class="mt-2 d-flex flex-column">
            <button class="btn btn-primary mx-3" @click="$router.push({name: 'Entrada', params: {id:'nuevo'}})">
                <i class="fa fa-plus-circle"></i> Nueva entrada
            </button>
        </div>

        <div class="entry-scrollarea">
            <entrada 
                v-for="entrada in entradasPorTermino" 
                :key="entrada.id" 
                :entrada="entrada"></entrada>
        </div>
    </div>
</template>

<script>
import {defineAsyncComponent } from 'vue'
import {mapGetters} from 'vuex'

export default {
    components: {
        Entrada: defineAsyncComponent (()=>import('@/modulos/diario/components/Entradas.vue'))
    },
    computed: {
        ...mapGetters ('diario',['obtenerEntradaBySearch']),
        entradasPorTermino(){
            return this.obtenerEntradaBySearch (this.term)
        }
    },
    data(){
        return {
            term: ''
        }
    }
}
</script>

<style lang="scss" scoped>

    .entry-list-container{
        border-right: 1px solid #2c3e50;
        height: calc(100vh - 56px);
    }
    .entry-scrollarea{
        height: calc(100vh - 110px);
        overflow: scroll;
    }
</style>
