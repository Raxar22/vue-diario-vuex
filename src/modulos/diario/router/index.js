


export default {

    name: 'diario',
    component: () => import(/* webpackChunkName: "diario" */ '@/modulos/diario/layouts/DiarioLayout.vue' ),
    children: [
        {
            path:'',
            name:'NoEntradas',
            component: () => import (/* webpackChunkName: "SinEntradas" */ '@/modulos/diario/views/NingunaEntradaSeleccionada.vue'),
        },
        {
            path:':id',
            name:'Entrada',
            component: () => import (/* webpackChunkName: "Entradas" */ '@/modulos/diario/views/EntradaDiario.vue'),
            props: (route) =>{
                return {
                    id: route.params.id
                }
            }

        }
    ]


}