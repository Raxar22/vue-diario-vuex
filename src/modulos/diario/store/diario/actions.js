
// export const miAccion = async ({commit})=> {

import diarioApi from "@/api/diarioApi"

// }

export const cargarEntradas = async ( {commit} )=> {

    const {data} = await diarioApi.get('/Entradas.json')
    if (!data){
        commit('setEntrada',[])
        return
    }
    const entradas = []
    for (let id of Object.keys(data)){
        entradas.push({
            id,
            ...data[id]
        })
    }
    commit('setEntrada',entradas)
   // console.log (entradas)
}

export const actualizarEntradas = async ( {commit}, entrada )=> {
    //dispara la mutation
    const {fecha,text,imagen} = entrada
    const datoAGuardar =  {fecha,text,imagen}
    await diarioApi.put(`/Entradas/${entrada.id}.json`,datoAGuardar)
    datoAGuardar.id = entrada.id
    commit('updateEntrada',{...datoAGuardar})


}

export const crearEntradas = async ( {commit},entrada )=> {
    const {fecha,text,imagen} = entrada
    const datoAGuardar =  {fecha,text,imagen}
    const entradaNueva = await diarioApi.post(`/Entradas.json`,datoAGuardar)

    datoAGuardar.id = entradaNueva.data.name
    //console.log(entradaNueva.data.name)
    commit('addEntrada',{...datoAGuardar})

    return entradaNueva.data.name

}


export const borrarEntrada = async ( {commit},id )=> {

    await diarioApi.delete(`/Entradas/${id}.json`)
    commit('borrarEntrada',id)
}