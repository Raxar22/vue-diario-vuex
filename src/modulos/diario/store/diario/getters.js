
// export const miGetter = (state)=> {
//  return state
// }

export const obtenerEntradaBySearch = ( state )=> (term = '') => {

    if (term.length===0) return state.entries

    return state.entries.filter (entry => entry.text.toLowerCase().includes(term.toLowerCase()))
}


export const obtenerEntradaById = (state)=> (id='') => {
    const entrada= state.entries.find (entry=> entry.id === id)
    
    if (!entrada) return
    
    return {...entrada}
}