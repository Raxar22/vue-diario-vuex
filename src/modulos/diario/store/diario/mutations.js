
// export const miMutacion =  (state)=> {

// }

export const setEntrada =  (state,entradas)=> {
    state.entries= []
    state.entries= [...state.entries, ...entradas]
    state.isLoading = false
}

export const updateEntrada =  (state,entrada)=> {
    const indice = state.entries.map( e => e.id ).indexOf(entrada.id)
    state.entries[indice] = entrada

}

export const addEntrada =  (state,entrada)=> {
    // console.log(state)
    // console.log(entrada)
    state.entries = [entrada, ...state.entries]
}
export const borrarEntrada =  (state,id)=> {
  /*  const indice = state.entries.map( e => e.id ).indexOf(id)
    state.entries.splice(indice,1)*/


    state.entries = state.entries.filter (entrada => entrada.id !== id)

}