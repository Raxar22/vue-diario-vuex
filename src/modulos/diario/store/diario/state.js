
export default ()  => ({
    isLoading : true,
    entries: [
        {
            id: new Date().getTime(),
            date: new Date().toDateString(),
            text: 'Voluptate laborum ex sit labore duis excepteur qui adipisicing. Mollit proident qui anim tempor minim sunt ea sint. Elit enim ad et velit mollit eiusmod nostrud aute tempor nostrud consequat nisi.',
            picture: null,
        },
        {
            id: new Date().getTime()+1000,
            date: new Date().toDateString(),
            text: 'Qui Lorem exercitation commodo proident labore consequat mollit nostrud minim. Sit culpa ex voluptate exercitation ea elit pariatur anim in consequat. Ex consectetur non irure deserunt ad incididunt qui nisi. Sint duis cupidatat reprehenderit id ipsum ullamco amet ex.',
            picture: null,
        },
        {
            id: new Date().getTime()+2000,
            date: new Date().toDateString(),
            text: 'Qui cupidatat labore magna consectetur. Mollit dolore laboris ut non culpa enim ipsum tempor. Eiusmod est minim mollit exercitation. Est duis Lorem consectetur enim occaecat duis duis enim deserunt ipsum cupidatat commodo velit deserunt. Culpa aliqua deserunt officia mollit eu proident. Minim aliquip esse ex quis aliquip anim enim anim esse laboris.',
            picture: null,
        },
    ]
})