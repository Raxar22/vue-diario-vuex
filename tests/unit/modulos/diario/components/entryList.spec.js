import { shallowMount } from "@vue/test-utils"
import { createStore } from "vuex"
import diario from '@/modulos/diario/store/diario/index'
import EntryList from '@/modulos/diario/components/EntryList'
import { diarioState } from "../../../Mocks/test-diario-state"

const createVuexStore = (inicialState)=> createStore({
    modules:{
        diario: {
            ...diario,
            state: {...inicialState}
        }
    }
})  

describe('Pruebas en el EntryList',()=>{
    
    const store = createVuexStore(diarioState)
    const mockRouter = {
        push: jest.fn()
    }


    let wrapper

    beforeEach(()=>{
        jest.clearAllMocks()
        wrapper = shallowMount (EntryList, {
            global:{
                mocks:{
                    $router: mockRouter
                },
                plugins: [store]
            }
        })
    })

    test('Debe llamar entradasPorTermino y sacar dos entradas',()=>{

        expect (wrapper.findAll('entrada-stub').length).toBe(2)
        expect (wrapper.html()).toMatchSnapshot()

    })

    test ('Debe llamar el entradasPorTermino y filtrar las entradas',async ()=>{
        const input = wrapper.find('input')
        await input.setValue('segunda')
        expect (wrapper.findAll('entrada-stub').length).toBe(1)
    })
    test ('El boton Nuevo debe redireccionar a /new',()=>{
        wrapper.find('button').trigger('click')
        expect(mockRouter.push).toHaveBeenCalledWith({name: 'Entrada', params: {id:'nuevo'}})
    })

})