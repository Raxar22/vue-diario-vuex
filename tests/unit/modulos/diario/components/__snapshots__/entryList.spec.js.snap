// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryList Debe llamar entradasPorTermino y sacar dos entradas 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Buscar Entrada"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> Nueva entrada </button></div>
  <div class="entry-scrollarea">
    <entrada-stub entrada="[object Object]"></entrada-stub>
    <entrada-stub entrada="[object Object]"></entrada-stub>
  </div>
</div>
`;
